OSDOCS-18179-Bare-Metal-Networking-tavery:Bug Batching (#112345)
openshift/openshift-docs: modules/rn-ocp-release-notes-fixed-issues.adoc

//[id="rn-ocp-release-note-api-auth-bug-fixes_{context}"]
//== API Server and Authentication

[id="rn-ocp-release-note-bare-metal-hardware-fixed-issues_{context}"]
== Bare Metal Hardware Provisioning

> * Before this update, when provisioning an NVIDIA DGX B200 bare-metal node using Advanced Cluster Management, the `BareMetalHost` could get stuck in a "Provisioning" state. With this release, the credential detection has been updated to detect missing credentials errors that were not being detected previously. (link:https://redhat.atlassian.net/browse/OCPBUGS-62309[OCPBUGS-62309])

Reviewer: ```suggestion
* Before this update, when provisioning an NVIDIA DGX B200 bare-metal node using Advanced Cluster Management, the `BareMetalHost` object could get stuck in a `Provisioning` state. With this release, the credential detection has been updated to detect missing credentials errors that were not being detected previously. (link:https://redhat.atlassian.net/browse/OCPBUGS-62309[OCPBUGS-62309])
```

Reviewer: I have no idea if that is the correct noun. I did a quick search and that came up. I'd double check my findings. After reading the next bug, should this be the general usage bare-metal host (BMH)

Author: Yes, I changed that instance to "bare-metal host (BMH)".

> * Before this update, during the cluster deletion got stuck during the inspection phase due to a power off stage transition. As a consequence, the cluster was not deleted. With this release, the bare-metal host (BMH) is prevented from getting stuck during deletion in a ZTP environment. As a result, the cluster removal is prevented from getting stuck during the inspection phase, which improves the efficiency of the ZTP environment. (link:https://redhat.atlassian.net/browse/OCPBUGS-65571[OCPBUGS-65571])

Reviewer: ```suggestion
* Before this update, the cluster deletion got stuck during the inspection phase due to a power off stage transition. As a consequence, the cluster was not deleted. With this release, the bare-metal host (BMH) is prevented from getting stuck during deletion in a ZTP environment. (link:https://redhat.atlassian.net/browse/OCPBUGS-65571[OCPBUGS-65571])
```

* Before this update, the resource ID for physical network interface controllers (NICs) on bare-metal machines using the HPE iLO6 system could change unpredictably when rebooting the machine. With this release, the resource ID stays the same after reboots. (link:https://redhat.atlassian.net/browse/OCPBUGS-70226[OCPBUGS-70226])

* Before this update, when performing a firmware update on a bare-metal machine by updating the `HostFirmwareController` configuration, if the `HostUpdatePolicy` was set to `onReboot`, the Bare Metal Operator (BMO) would sometimes fail to initiate the reboot and firmware upgrade. With this release, the BMO initiates the reboot and performs the firmware upgrade. (link:https://redhat.atlassian.net/browse/OCPBUGS-75006[OCPBUGS-75006])

> * Before this update, when a physical network interface was enslaved to a bridge, for example, OVN-Kubernetes `br-ex`, the interface and bridge shares the same MAC address. As a consequence, the provisioning interface detection would match both the interface and bridge and concatenate their names into an invalid multi-line value, whicn prevented the ironic service from starting. Because the ironic service could not start, workers could not PXE boot and agent-based installation failed with `SyncingFailed`. With this release, the bash text-parsing logic is replaced with a Python script using the `ip -json -d` command for structured output, which correctly selects a single interface when there are multiple matches for a MAC or IP address. As a result, agent-based installation on a bare-metal node completes successfully in bridged network configurations. (link:https://redhat.atlassian.net/browse/OCPBUGS-77528[OCPBUGS-77528])

Reviewer: ```suggestion
* Before this update, when a physical network interface was enslaved to a bridge the interface and bridge shared the same MAC address. As a consequence, the provisioning interface detection would match both the interface and bridge and concatenate their names into an invalid multi-line value, which prevented the ironic service from starting. Because the ironic service could not start, workers could not PXE boot and agent-based installation failed with `SyncingFailed`. With this release, the bash text-parsing logic is replaced with a Python script using the `ip -json -d` command for structured output, which correctly selects a single interface when there are multiple matches for a MAC or IP address. As a result, agent-based installation on a bare-metal node completes successfully in bridged network configurations. (link:https://redhat.atlassian.net/browse/OCPBUGS-77528[OCPBUGS-77528])
```

> * Before this update, when provisioning a bare-metal machine, if you provided a network data Secret that was missing an `nmstate` key, the network data would be discarded silently, leading to unexpected errors. With this release, if the `nmstate` key is missing, the Image Customization Controller will generate an error informing that the `nmstate` key is required to complete provisioning. (link:https://redhat.atlassian.net/browse/OCPBUGS-77840[OCPBUGS-77840])

Reviewer: ```suggestion
* Before this update, when provisioning a bare-metal machine, if you provided a network data Secret that was missing an `nmstate` key, the network data would be discarded silently, leading to unexpected errors. With this release, if the `nmstate` key is missing, the Image Customization Controller will generate an error informing you that the `nmstate` key is required to complete provisioning. (link:https://redhat.atlassian.net/browse/OCPBUGS-77840[OCPBUGS-77840])
```

> * Before this update, the bootstrap machine created during a bare-metal installation did not have a serial console log file. If the bootstrap machine failed and SSH access was not available, the logs were inaccessible. With this release, the bootstrap machine creates a serial console log file that is accessible to must-gather. (link:https://redhat.atlassian.net/browse/OCPBUGS-78589[OCPBUGS-78589])

Reviewer: ```suggestion
* Before this update, the bootstrap machine created during a bare-metal installation did not have a serial console log file. If the bootstrap machine failed and SSH access was not available, the logs were inaccessible. With this release, the bootstrap machine creates a serial console log file that is accessible to `must-gather` tool. (link:https://redhat.atlassian.net/browse/OCPBUGS-78589[OCPBUGS-78589])
```

Reviewer: This should be checked.

Author: It is referenced as "must-gather diagnostic tool" without backticks in https://docs.redhat.com/en/documentation/red_hat_ai_inference/3.4/html/deploy_distributed_inference_with_llm-d_on_azure_or_coreweave_kubernetes_service/running-must-gather-on-xks_llmd-on-xks

// [id="rn-ocp-release-note-kube-api-server-fixed-issues_{context}"]

> // == Kubernetes API Server* Before this update, metrics for the Cluster Baremetal Operator were exposed on all network interfaces on a bare-metal host. With this release, the metrics are only exposed to the `localhost` interface so that the `kube-rbac-proxy` sidecar can access the metrics. (link:https://redhat.atlassian.net/browse/OCPBUGS-84924[OCPBUGS-84924])

Reviewer: ```suggestion
== Kubernetes API Server 
* Before this update, metrics for the Cluster Baremetal Operator were exposed on all network interfaces on a bare-metal host. With this release, the metrics are only exposed to the `localhost` interface so that the `kube-rbac-proxy` sidecar can access the metrics. (link:https://redhat.atlassian.net/browse/OCPBUGS-84924[OCPBUGS-84924])
```

Reviewer: This is the same bug that is the above batch. This needs investigated. Is this a sloppy copy and paste?

Author: Yes, this is a sloppy copy and paste. I deleted the bug from this section.
